src/sp/it/pl/util/file/Util.kt

```diff
@@ -33,6 +33,9 @@ val File.nameOrRoot: String get() = name.takeUnless { it.isEmpty() } ?: toString
 // TODO: does not work for directories with '.'
 val File.nameWithoutExtensionOrRoot: String get() = nameWithoutExtension.takeUnless { it.isEmpty() } ?: toString()
 
+val File.isPlayable
+    get() = AudioFileFormat.of(toURI()).isSupported(AudioFileFormat.Use.PLAYBACK)
```

Comment:
But the contextmenu needs the `isPlayable` property to generate extra entries where appropriate, so it needs to be accessible. We could rename it to something like `isMusic`, but this method might also be required for implementing some widgets in the future.

Reply:
isMusic makes no sense here as it is a general concept. It needs to be isPlayable, where playability means exactly 'can be played by this application'. Considering the utils as a mere dependency, possibly shared in other projects, isPlayable has no right to exist there.

Question now becomes whether it should be a service method isPlayable(file) or an extension method defined somewhere in the project. The problem with the former is the annoying way this is invoked (not on the instance of a File but the service itself), the problem with latter the hidden fact that this method only works in the context of the application. Well, since there is no dependence on application state, it can be a top level function and hence an extension function.

But there may not be only single isPlayable() concept in an application, which again raises the question how much does an application-scope concept like isPlayable make sense as an extension method. The truth is not a whole lot.

An alternative would be to have this check provided as 'Item -> Boolean' by the service (some PlaybackManager or something), perhaps even just in companion object to allow 'static import', so the call would then become some form of file.let(isPlayable).

Reply:
One of the reasons the utils is tightly coupled to the core module are these application util methods and general purpose methods mixed up together in the utils. This is not a coincidence however, the utils were never intended to be taken out of the project. Until now. Separating them now involves identifying all of the application scoped functionality and moving it back to application-util.

This would increase fragmentation of the code base, which is why I was against the decoupling in the first place.

Reply:
As we agreed, we leave isPlayable where you put it and the issue will be taken care of in the future along mime types and file format handling in general